distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso

```diff
@@ -1049,15 +1048,16 @@ type Column
          supported type is chosen and a `Inexact_Type_Coercion` problem is
          reported.
     cast : Value_Type -> Problem_Behavior -> Column ! Illegal_Argument | Inexact_Type_Coercion | Lossy_Conversion
-    cast self value_type=self.value_type on_problems=Problem_Behavior.Report_Warning =
-        dialect = self.connection.dialect
-        type_mapping = dialect.get_type_mapping
-        target_sql_type = type_mapping.value_type_to_sql value_type on_problems
-        target_sql_type.if_not_error <|
-            infer_from_database new_expression =
-                SQL_Type_Reference.new self.connection self.context new_expression
-            new_column = dialect.make_cast self.as_internal target_sql_type infer_from_database
-            Column.Value new_column.name self.connection new_column.sql_type_reference new_column.expression self.context
+    cast self value_type=Value_Type.Char on_problems=Problem_Behavior.Report_Warning =
+        check_cast_compatibility self.value_type value_type . if_not_error <|
```

Comment:
I guess you are right. I somehow preferred `if_not_error` pattern, because it is more pattern-y - it is searchable etc.
But I guess taking a suspended action is the right choice here.

Reply:
Thanks for pointing this out - turns out I had a bug there and I only noticed it when refactoring this method.
Also moving into suspended `action` argument is exactly what helped me spot this bug, so I think this proves that that pattern is indeed safer in cases like this one.